### ISSUE TITLE
Why there is no `unapplySeq` in `Chain`?

### ISSUE THREAD

**Issue by taeguk** [typelevel/cats#2960]
I found there is no `unapplySeq` in `Chain`.
So some pattern matchings can't be compiled:
```scala
val c = Chain(1, 2, 3)
c match {
  case Chain(_, _, _) => ???  // can't be compiled because there is no `unapplySeq` in `Chain`
}
```

Fortunately, there is workaround:
```scala
val c = Chain(1, 2, 3)
c match {
  case ChainPattern(_, _, _) => ??? 
}

object ChainPattern {
  def unapplySeq[A](chain: Chain[A]): Option[Seq[A]] =
    Some(chain.toList)
}
```
But, it is bothersome.

So, I want to know why there is no `unapplySeq` in `Chain`.

**Comment by kailuowang** [comment]
I don't think there is any particular reason other than no one ever got to it until...You! 
Wanna try submit a PR? Some sort of uncon unapply would be great too!
```scala
object ==: {
  def unapply[T](c: Chain[T]): Option[(T, Chain[T])] =
    c.uncons
}
```
so that you can pattern with `head ==: tail`
`Not sure if `==:` is the best symbol to use here. Ideally it should be something that resemble a chain?

**Comment by vasiliybondarenko** [comment]
I've just added PR for this issue

**Comment by taeguk** [comment]
This is fixed in #2990
